### PR TITLE
extconf.rb: locate zstd when using homebrew

### PR DESCRIPTION
**Pull request by byroot**
Fix compilation issues on my system.

cc @tenderlove

### CODE REVIEW

File: ext/mysql2/extconf.rb [hunk 1]
```diff
@@ -66,6 +66,11 @@ def add_ssl_defines(header)
   $LDFLAGS << " -L#{openssl_location}/lib" if openssl_location
 end
 
+if RUBY_PLATFORM =~ /darwin/ && system('command -v brew')
+  zstd_location = `brew --prefix zstd`.strip
+  $LIBPATH << "#{zstd_location}/lib"
```
Comment (sodabrew) [inline ext/mysql2/extconf.rb:71]: If zstd is not installed (perhaps for an older version of mysqlclient) I think `$LIBPATH` should not be modified. How about this:
```suggestion
  $LIBPATH << "#{zstd_location}/lib" if zstd_location
```

File: ext/mysql2/extconf.rb [hunk 2]
```diff
@@ -66,6 +66,12 @@ def add_ssl_defines(header)
   $LDFLAGS << " -L#{openssl_location}/lib" if openssl_location
 end
 
+if RUBY_PLATFORM =~ /darwin/ && system('command -v brew')
+  zstd_location = `brew --prefix zstd`.strip
+
+  $LIBPATH << "#{zstd_location}/lib" unless zstd_location.empty?
```
Comment (sodabrew) [inline ext/mysql2/extconf.rb:72]: If you found `.empty?` necessary vs. falsiness of the empty string, then the `if openssl_location` clause above should change to match -- did you test to see if brew's output when the package is not installed is in fact "truthy"?

```suggestion
  $LIBPATH << "-L#{zstd_location}/lib" unless zstd_location.empty?
```
Also I notice the openssl pattern above adds `-L` -- I think these ought to match? But if I'm missing something let me know!

Comment (byroot) [reply ext/mysql2/extconf.rb:72]: > vs. falsiness of the empty string

Empty strings aren't falsy in Ruby.

> Also I notice the openssl pattern above adds -L -- I think these ought to match? 

They're not adding to the same variables. `$LIBPATH` takes paths, `$LDFLAGS` takes flags. I guess I can rewrite the OpenSSL one to use LIBPATH and properly do its error checks.

Comment (byroot) [reply ext/mysql2/extconf.rb:72]: Done.

Comment (sodabrew) [reply ext/mysql2/extconf.rb:72]: 🤦 quite right. So what's that line above testing? Seems like a bogon, unless there's some long-lost mystery behavior of yore (old code, after all)

**Review by sodabrew — APPROVED** [review]
Thank you! Looking consistent and sensible now

File: ext/mysql2/extconf.rb (hunk 2 above)
Comment (byroot) [reply ext/mysql2/extconf.rb:72]: It wasn't testing anything. Was likely just an oversight.

File: ext/mysql2/extconf.rb [hunk 3]
```diff
@@ -63,7 +63,13 @@ def add_ssl_defines(header)
 # Homebrew OpenSSL on MacOS
 elsif RUBY_PLATFORM =~ /darwin/ && system('command -v brew')
   openssl_location = `brew --prefix openssl`.strip
-  $LDFLAGS << " -L#{openssl_location}/lib" if openssl_location
+  $LIBPATH << "#{openssl_location}/lib" unless openssl_location.empty?
+end
+
+if RUBY_PLATFORM =~ /darwin/ && system('command -v brew')
+  zstd_location = `brew --prefix zstd`.strip
+
```
Comment (sodabrew) [inline ext/mysql2/extconf.rb:71]:
```suggestion
```

**Pull request merged**